src/libraries/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray_1.Builder.cs

```diff
@@ -933,11 +933,7 @@ public void Sort(Comparison<T> comparison)
 
                 if (Count > 1)
                 {
-                    // Array.Sort does not have an overload that takes both bounds and a Comparison.
-                    // We could special case _count == _elements.Length in order to try to avoid
-                    // the IComparer allocation, but the Array.Sort overload that takes a Comparison
-                    // allocates such an IComparer internally, anyway.
-                    Array.Sort(_elements, 0, _count, Comparer<T>.Create(comparison));
+                    Array.Sort(_elements, comparison);
```

Comment:
Won't this break due to sorting the whole array, not just `_count` elements?

Reply:
D'oh! Yes, thanks for catching that.

Reply:
Oh, wait, that overload doesn't exist.

Reply:
> Oh, wait, that overload doesn't exist.

Right, that's what the comment you deleted said :-)

Reply:
I have read and grokked the comment, and now we sort only `_count` elements.